src/bp/core/services/telemetry/actions.ts

```diff
@@ -0,0 +1,111 @@
+import { parseActionInstruction } from 'common/action'
+import { BUILTIN_MODULES } from 'common/defaults'
+import LicensingService from 'common/licensing-service'
+import { getSchema } from 'common/telemetry'
+import Database from 'core/database'
+import { calculateHash } from 'core/misc/utils'
+import { TelemetryRepository } from 'core/repositories/telemetry_payload'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import ms from 'ms'
+import path from 'path'
+
+import { GhostService } from '..'
+import { JobService } from '../job-service'
+
+import { TelemetryStats } from './telemetry-stats'
+
+type NextNode = {
+  condition: string
+  node: string
+}
+
+type Node = {
+  id: string
+  name: string
+  next: Array<NextNode>
+  onEnter: Array<string>
+  onReceive: Array<string>
+  type: string
+}
+
+type Flow = {
+  flowName: string
+  botID: string
+  actions: Array<string>
+}
+
+@injectable()
+export class ActionsStats extends TelemetryStats {
+  protected url: string
+  protected lock: string
+  protected interval: number
+
+  constructor(
+    @inject(TYPES.GhostService) ghostService: GhostService,
+    @inject(TYPES.Database) database: Database,
+    @inject(TYPES.LicensingService) licenseService: LicensingService,
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-actions'
+    this.interval = ms('1d')
+  }
+
+  protected async getStats() {
+    return {
+      ...getSchema(await this.getServerStats(), 'server'),
+      event_type: 'builtin_actions',
+      event_data: { schema: '1.0.0', flows: await this.getFlowsWithActions() }
+    }
+  }
+
+  private async getFlowsWithActions() {
```

Comment:
Missing typings for the return value. You even return either an array or an object, typings help prevent this kind of mistake